src/Compilers/CSharp/Portable/Errors/CSharpDiagnosticFormatter.cs

```diff
@@ -11,5 +11,10 @@ internal CSharpDiagnosticFormatter()
         }
 
         public static new CSharpDiagnosticFormatter Instance { get; } = new CSharpDiagnosticFormatter();
+
+        internal override bool HasDefaultHelpLinkUri(Diagnostic diagnostic)
+        {
+            return diagnostic.Descriptor.HelpLinkUri == ErrorFacts.GetHelpLink((ErrorCode)diagnostic.Code);
```

Comment:
Did you consider exposing a property on DiagnosticDescriptor to get "the help link URI, if a custom/'meaningful' one was specified"?

Reply:
> because they clone DiagnosticDescriptors

I have to wonder why they do. Even though my instincts say, don't ask, because then you'll know. ;)

I think if we have to assume any new state needs a public accessor and public constructor parameter, in order to support "cloning" it ad-hoc, then your solution is probably best.